docs/Architecture.md

```diff
@@ -1,49 +1,93 @@
+  - As an [OLM operator](https://olm.operatorframework.io/), it is designed with `operator-sdk`, and allows subscriptions for easy updates.
+- [A CLI](https://github.com/netobserv/network-observability-cli) that also manages some of the above components, for on-demand monitoring and packet capture.
+  - It is provided as a `kubectl` or `oc` plugin, allowing to capture flows (similar to what the operator does, except it's on-demand and in the terminal) or full packets (much like a `tcpdump` command).
+  - It is also available via [Krew](https://krew.sigs.k8s.io/).
+  - Check out the blog post: [Network observability on demand ](https://developers.redhat.com/articles/2024/09/17/network-observability-demand#what_is_the_network_observability_cli_).
+
+## Direct deployment model
+
+When using the operator with `FlowCollector` `spec.deploymentModel` is set to `Direct`, agents and FLP are both deployed per node (as `DaemonSets`). This is perfect for an assessment of the technology, suitable on small clusters, but isn't very memory efficient in large clusters as every instance of FLP ends up caching the same cluster information, which can be huge.
+
+Note that Loki isn't managed by the operator and must be installed separately, such as with the Loki operator. Same goes with Prometheus and any custom receiver.
+
+<!-- You can use https://mermaid.live/ to test it -->
+
+```mermaid
+flowchart TD
+    subgraph "for each node"
+        A[eBPF Agent] -->|generates flows| F[FLP]
+    end
+    F -. exports .-> E[(Kafka/Otlp/IPFIX)]
+    F -->|raw logs| L[(Loki)]
+    F -->|metrics| P[(Prometheus)]
+    C[Console plugin] <-->|fetches| L
+    C <-->|fetches| P
+    O[Operator] -->|manages| A
+    O -->|manages| F
+    O -->|manages| C
+```
+
+## Kafka deployment model
+
+When using the operator with `FlowCollector` `spec.deploymentModel` is set to `Kafka`, only the agents are deployed per node as a `DaemonSet`. FLP becomes a Kafka consumer that can be scaled independently. This is the recommended mode for large clusters, and is a more robust/resilient solution.
+
+Like in `Direct` mode, data stores aren't managed by the operator. The same applies to the Kafka brokers and stores. You can check the Strimzi operator for that.
+
+<!-- You can use https://mermaid.live/ to test it -->
+
+```mermaid
+flowchart TD
+    subgraph "for each node"
+        A[eBPF Agent]
+    end
+    A -->|produces flows| K[(Kafka)]
+    F[FLP] <-->|consumes| K
+    F -. exports .-> E[(Kafka/Otlp/IPFIX)]
+    F -->|raw logs| L[(Loki)]
+    F -->|metrics| P[(Prometheus)]
+    C[Console plugin] <-->|fetches| L
+    C <-->|fetches| P
+    O[Operator] -->|manages| A
+    O -->|manages| F
+    O -->|manages| C
+```
+
+## CLI
+
+When using the CLI, the operator is not involved, which means you can use it without installing NetObserv as a whole. It uses a special mode of the eBPF agents that embeds FLP.
```

Comment:
cc @jpinsonneau does it look good to you? cf https://github.com/netobserv/network-observability-operator/blob/e80a75f18fa0f4c28c1819311415d735038206dd/docs/Architecture.md#cli

Reply:
Yes sounds good. Note that the collector is not involved for the metrics generation. That could be mentionned too

Reply:
```suggestion
When using the CLI, the operator is not involved, which means you can use it without installing NetObserv as a whole. It uses a special mode of the eBPF agents that embeds FLP.

Flows and Packets captures will deploy both agents and collector.
Metrics captures will only deploy agents.
```

Reply:
Maybe that's something I missed .. how does metrics work when using the kubectl plugin ? The readme mentions only a make target for capturing metrics. If I run `oc netobserv flows ...` can I get metrics as well?

Reply:
nvm I've checked the PR, it's explained: `$ ./build/oc-netobserv metrics --enable_pktdrop="true" --enable_dns="true" --enable_rtt="true"`

next question: README says that's only for openshift, but what if I have prometheus without openshift? I guess I can still have the metrics, it's just the charts that I won't get, correct?

Reply:
I added details in last commit https://github.com/netobserv/network-observability-operator/pull/1022/commits/c51d0d754fa9df89f9a9abeeaa3df7a3189279d8

Reply:
Yes if you have a ServiceMonitor you will have the metrics
But the configmap in *openshift* namespace will be ignored